### PR TITLE
Prevent users from upvoting Qs more than once

### PR DESCRIPTION
**Pull request by j-blackmore**
Users can now only upvote a question once.
- We store the user votes in a upvotes String array inside each question
- Store a vote as the user_id retrieved from auth0
- Now display number of votes as the count of entries in the array i.e. `question.upvotes.length`
- The 'thumbs up' upvote button, now changes to red when upvoted.

### CODE REVIEW

File: server/models/question.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ const questionSchema = new Schema({
     author: String,
     role: Boolean,
     datetime: Date,
-    upvotes: Number,
+    upvotes: [String],
```
Comment (JacobCrozier) [inline server/models/question.js]: How come? Very interesting solution

File: server/routes/questionRoute.js [hunk 2]
```diff
@@ -10,36 +10,40 @@ router.get('/', (req, res) => {
 });
 
 router.get('/:id', (req, res) => {
-    Question.find({_id: req.params.id})
+    Question.find({ _id: req.params.id })
```
Comment (JacobCrozier) [inline server/routes/questionRoute.js]: Well spotted

File: server/models/question.js (hunk 1 above)
Comment (j-blackmore) [reply server/models/question.js]: So we are now storing a list of user Id's in the upvotes, which just represent a user has upvoted the question

File: src/auth/Auth.js [hunk 3]
```diff
@@ -37,6 +37,10 @@ export default class Auth {
             authResult.expiresIn * 1000 + new Date().getTime()
         );
 
+        this.auth0.client.userInfo(authResult.accessToken, function(err, user) {
+            localStorage.setItem('user', JSON.stringify(user));
+        });
+
```
Comment (JacobCrozier) [inline src/auth/Auth.js]: arrow () => ?

File: src/components/Question/EnterQuestionBox.js [hunk 4]
```diff
@@ -29,7 +29,7 @@ export default class EnterQuestionBox extends React.Component {
                 question_title: "",
                 author: "",
                 datetime: null,
-                upvotes: 0,
+                upvotes: [],
```
Comment (JacobCrozier) [inline src/components/Question/EnterQuestionBox.js]: sublime

File: server/models/question.js (hunk 1 above)
Comment (JacobCrozier) [reply server/models/question.js]: Nice

File: src/auth/Auth.js (hunk 3 above)
Comment (j-blackmore) [reply src/auth/Auth.js]: oh yes, will change

**Pull request merged**